### PR TITLE
Add try catch when current login user doesn't have permission to assign a role.

### PR DESCRIPTION
**Pull request by LiliDeng**
Test under 3 situations.
1 Normal case - when login user can create app and assign role successfully.
2 When login user can create app but doesn't have permission to assign a role, and she chose to delete the app.
3 When login user can create app but doesn't have permission to assign a role, and she chose not to delete the app.

### CODE REVIEW

File: Utilities/CreateServicePrincipal.ps1 [hunk 1]
```diff
@@ -223,21 +223,40 @@ function New-ServicePrincipal() {
...
+        }
     }
 
-    Write-Host "Successfully created Service Principal`n"
     Write-Host "==============Created Serivce Principal=============="
     Write-Host "SUBSCRIPTION_ID:" $subscription.Id
     Write-Host "CLIENT_ID:      " $clientId
     Write-Host "TENANT_ID:      " $tenantId
     Write-Host "CLIENT_SECRET:  " $clientSecret
+
```
Comment (juhlee-microsoft) [inline Utilities/CreateServicePrincipal.ps1]: delete a line

File: Utilities/CreateServicePrincipal.ps1 [hunk 2]
```diff
@@ -30,17 +30,17 @@ function New-SWRandomPassword {
...
         3ABa
 
-        Generates a password with a length of 4 containing atleast one char from each InputString that will start with a letter from
+        Generates a password with a length of 4 containing at least one char from each InputString that will start with a letter from
         the string specified with the parameter FirstChar
     .OUTPUTS
         [String]
     .NOTES
-        Written by Simon Wåhlin, blog.simonw.se
+        Written by Simon Wahlin, blog.simonw.se
```
Comment (juhlee-microsoft) [inline Utilities/CreateServicePrincipal.ps1:43]: Wait, if this comes from someone else's script, I am not sure if we put Apache license. Need to check with CELA.

File: Utilities/CreateServicePrincipal.ps1 [hunk 3]
```diff
@@ -223,16 +223,34 @@ function New-ServicePrincipal() {
...
+    try {
+        if ($Privilege -eq 2) {
+            New-AzRoleAssignment -RoleDefinitionName "Owner" -ApplicationId $ClientId
+        }
+        else {
+            New-AzRoleAssignment -RoleDefinitionName "Contributor" -ApplicationId $ClientId
+        }
+        Write-Host "Successfully created Service Principal and assigned role...`n"
+    } catch {
+        $ErrorMessage = $_.Exception.Message
```
Comment (juhlee-microsoft) [inline Utilities/CreateServicePrincipal.ps1]: We have common rule printing out the error message with the error line number. Please refer to another place of 'catch block.

File: Utilities/CreateServicePrincipal.ps1 [hunk 4]
```diff
@@ -223,16 +223,34 @@ function New-ServicePrincipal() {
...
+        Write-Host -ForegroundColor Red "$ErrorMessage"
+        $currentSignInAccount = Get-AzContext | Select-Object Account
+        $role = Get-AzRoleAssignment -SignInName $currentSignInAccount.Account.id | Select-Object RoleDefinitionName
+        # https://docs.microsoft.com/en-us/powershell/azure/create-azure-service-principal-azureps?view=azps-2.5.0#manage-service-principal-roles
+        if ($ErrorMessage -match "does not have authorization to perform action 'Microsoft.Authorization/roleAssignments/write' over scope") {
```
Comment (juhlee-microsoft) [inline Utilities/CreateServicePrincipal.ps1]: Using 'message' text string in the conditional statement is not recommended. Can you switch to the error code?

File: Utilities/CreateServicePrincipal.ps1 [hunk 5]
```diff
@@ -223,16 +223,34 @@ function New-ServicePrincipal() {
...
+            Write-Host "You are the $($role.RoleDefinitionName) of subscription $($subscription.Name) - $($subscription.Id), don't have enough permission to assign a role."
+            $confirmation = Read-Host "Do you want to delete the service principal $identifier just created? [y/n]"
+            if($confirmation -eq "y") {
+                Remove-AzADApplication -ApplicationId $ClientId -Force
+                Write-Host "Delete $identifier successfully..."
+                return
+            } else {
+                Write-Host "Successfully created Service Principal but assigned role failed...`n"
```
Comment (juhlee-microsoft) [inline Utilities/CreateServicePrincipal.ps1]: What is the expected behavior of the user? I don't think we leave the intermediate artifact here.

**Review by juhlee-microsoft — CHANGES_REQUESTED** [review]

File: Utilities/CreateServicePrincipal.ps1 (hunk 4 above)
Comment (LiliDeng) [reply Utilities/CreateServicePrincipal.ps1]: I use $_

File: Utilities/CreateServicePrincipal.ps1 (hunk 5 above)
Comment (LiliDeng) [reply Utilities/CreateServicePrincipal.ps1]: I give the invoke person to make decision, remove it or keep it for further use, he can manually delete it.

Comment (juhlee-microsoft) [reply Utilities/CreateServicePrincipal.ps1]: Sorry for late comment. Please inform to the user as comment or logging likely saying the object was created but could not proceed for the right permission. Recommend to clean up manually.

**Review by juhlee-microsoft — CHANGES_REQUESTED** [review]

**Review by juhlee-microsoft — APPROVED** [review]

**Pull request merged**